lib/chef/resource/windows_script.rb

```
@@ -31,8 +31,8 @@ class WindowsScript < Chef::Resource::Script
       def initialize(name, run_context, resource_name, interpreter_command)
         super(name, run_context)
         @interpreter = interpreter_command
-        @resource_name = resource_name
-        @default_guard_interpreter = resource_name
+        @resource_name = resource_name if resource_name
```

Does this allow the silent failure you were trying to avoid above?